crates/net/network/src/transactions/fetcher.rs

```diff
@@ -101,203 +158,309 @@ impl TransactionFetcher {
-            return ValidTxHashes::empty_eth66()
+    pub(super) fn pack_hashes(
+        &mut self,
+        hashes_to_request: &mut RequestTxHashes,
+        hashes_from_announcement: ValidAnnouncementData,
+    ) -> RequestTxHashes {
+        if hashes_from_announcement.msg_version().is_eth68() {
+            return self.pack_hashes_eth68(hashes_to_request, hashes_from_announcement)
         }
-        let surplus_hashes = hashes.split_off(GET_POOLED_TRANSACTION_SOFT_LIMIT_NUM_HASHES - 1);
-
-        ValidTxHashes::new_eth66(surplus_hashes)
+        self.pack_hashes_eth66(hashes_to_request, hashes_from_announcement)
     }
 
-    /// Evaluates wether or not to include a hash in a `GetPooledTransactions` version eth68
-    /// request, based on the size of the transaction and the accumulated size of the
-    /// corresponding `PooledTransactions` response.
+    /// Packages hashes for a [`GetPooledTxRequest`] from an
+    /// [`Eth68`](reth_eth_wire::EthVersion::Eth68) announcement up to limit as defined by protocol
+    /// version 68. Takes a [`RequestTxHashes`] buffer as parameter for filling with hashes to
+    /// request.
     ///
-    /// Returns `true` if hash is included in request. If there is still space in the respective
-    /// response but not enough for the transaction of given hash, `false` is returned.
-    fn include_eth68_hash(&self, acc_size_response: &mut usize, hash: TxHash) -> bool {
-        debug_assert!(
-            self.eth68_meta.peek(&hash).is_some(),
-            "can't find eth68 metadata for `%hash` that should be of version eth68, broken invariant `@eth68_meta` and `@self`,
-`%hash`: {},
-`@self`: {:?}",
-            hash, self
-        );
+    /// Returns left over hashes.
+    ///
+    /// Loops through hashes passed as parameter and checks if a hash fits in the expected
+    /// response. If no, it's add to surplus hashes. If yes, it's add to hashes to request and
+    /// expected response size is accumulated.
+    pub(super) fn pack_hashes_eth68(
+        &mut self,
+        hashes_to_request: &mut RequestTxHashes,
+        hashes_from_announcement: ValidAnnouncementData,
+    ) -> RequestTxHashes {
+        let mut acc_size_response = 0;
+        let hashes_from_announcement_len = hashes_from_announcement.len();
 
-        if let Some(size) = self.eth68_meta.peek(&hash) {
-            let next_acc_size = *acc_size_response + size;
+        let mut hashes_from_announcement_iter = hashes_from_announcement.into_iter();
+
+        if let Some((hash, Some((_ty, size)))) = hashes_from_announcement_iter.next() {
+            hashes_to_request.push(hash);
+
+            // tx is really big, pack request with single tx
+            if size >= SOFT_LIMIT_BYTE_SIZE_POOLED_TRANSACTIONS_RESPONSE_MESSAGE {
+                return hashes_from_announcement_iter.collect::<RequestTxHashes>()
+            } else {
+                acc_size_response = size;
```

Comment:
is this an accumulator? if so, is this supposed to be `acc_size_response += size`?

Reply:
this is not called in a loop. the loop is below.

Reply:
you're right